Source/Core/DolphinQt/Config/Mapping/MappingCommon.cpp

```diff
@@ -63,33 +69,47 @@ class MappingProcessor : public QWidget
     if (!m_input_detector)
       return;
 
-    m_input_detector->Update(INPUT_DETECT_INITIAL_TIME, INPUT_DETECT_CONFIRMATION_TIME,
+    const auto confirmation_time =
+        INPUT_DETECT_CONFIRMATION_TIME * (m_parent->IsWaitForAlternateMappingsEnabled() ? 1 : 0);
+
+    m_input_detector->Update(INPUT_DETECT_INITIAL_TIME, confirmation_time,
                              INPUT_DETECT_MAXIMUM_TIME);
 
     if (m_input_detector->IsComplete())
     {
-      auto detections = m_input_detector->TakeResults();
-      ciface::MappingCommon::RemoveSpuriousTriggerCombinations(&detections);
+      auto* const button = m_clicked_mapping_buttons.back();
 
-      // No inputs detected. Cancel this and any other queued mappings.
-      if (detections.empty())
+      if (!FinalizeMapping(m_input_detector->TakeResults()))
       {
+        // No inputs detected. Cancel this and any other queued mappings.
         CancelMapping();
-        return;
       }
+      else if (m_parent->IsIterativeMappingEnabled() && m_clicked_mapping_buttons.empty())
+      {
+        button->QueueNextButtonMapping();
+      }
```

Comment:
Hmm... I don't know if this is something you can add cleanly, but the modifier really only exists for keyboard purposes.

Right now the iterative mapping feature includes expecting an input for the modifiers, which doesn't make much sense to assign for controllers.

Some ideas:
- Don't skip it if the device is the Keyboard/Mouse type and Skip it if the device is any other type
- Ignore Modifier entirely, since Keyboard already has a default binding preset so it's unlikely iterative mapping would be useful for KB, and even if the user wants to set Modifier they can after.
![image](https://github.com/user-attachments/assets/1994bccc-7bd6-4709-bf70-a492158eacc8)

Reply:
The `InputDetection` class has a `smoothness` value that hints whether the input is digital or analog. I can hook into that.

Reply:
I've now implemented this. Iterative input clears and skips the "Modifier" if an analog input was just mapped.

Reply:
This is a really clever solution!